### PR TITLE
zh-cn: update the translation of "Typical use cases of flexbox"

### PR DESCRIPTION
**Pull request by yin1999**
### Description

update the translation of "Typical use cases of flexbox"

### Related issues and pull requests

Part of: #14272

### CODE REVIEW

File: files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md [hunk 1]
```diff
@@ -109,28 +113,28 @@ You could also give each side different {{cssxref("flex-grow")}} factors, for ex
...
 
 {{EmbedGHLiveSample("css-examples/flexbox/use-cases/label-input-button.html", '100%', 550)}}
 
-Patterns like this can make it much easier to create a library of form elements for your design, which easily accommodate additional elements being added. You are taking advantage of the flexibility of flexbox by mixing items that do not grow with those that do.
+像这样的模式可以让你更容易地为你的设计创建一系列表单元素，这些元素可以轻松地适应额外的元素的添加。你可以将不可增长的元素和可增长的元素混合在一起以充分利用弹性盒的灵活性。
 
 ## 结论
 
-当了解上面的这些示例时，你已经满怀希望的想象怎么找到最好的方式使用弹性盒子布局实现你想要的结果。大部分情况下，你拥有不止一种选择。将可以变化的内容和不能变化的混合在一起，通过内容来决定他们的大小或者允许弹性布局设置按比例分配空间。要因地制宜，对症下药。
+在探索上述模式时，你已经满怀希望的想象怎么找到使用弹性盒子实现你想要的结果的最佳方式。大部分情况下，你拥有不止一种选择。将可以变化的内容和不能变化的元素混合在一起，通过内容来决定它们的大小或者允许弹性布局设置按比例分配空间。要因地制宜，对症下药。
```
Comment (awxiaoxian2020) [inline files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md:138]:
```suggestion
在探索上述模式时，你已经满怀希望地想象怎么找到使用弹性盒子实现你想要的结果的最佳方式。大部分情况下，你拥有不止一种选择。将可以变化的内容和不能变化的元素混合在一起，通过内容来决定它们的大小或者允许弹性布局设置按比例分配空间。要因地制宜，对症下药。
```

**Review by awxiaoxian2020 — CHANGES_REQUESTED** [review]

File: files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md [hunk 2]
```diff
@@ -1,89 +1,93 @@
...
 
-在下面的示例中，我们让各元素都展示为其自身的尺寸，通过使用 `justify-content: space-between` 使元素之间拥有相同的空间。你可以通过 `space-around` 的值来改变空间分布的方式，在浏览器支持的环境下还可以使用 `space-evenly`。你也可以使用 `flex-start` 让空间展示在所有元素末尾。使用 `flex-end` 让空间展示在所有元素之前， `center` 可以剧中所有元素。
+在下面的示例中，我们通过使用 `justify-content: space-between` 使元素之间拥有相同的空间，让各元素都以其自身的尺寸展示。你可以通过 `space-around`（或 `space-evenly`，如果支持）值来改变分布空间的方式。你也可以使用 `flex-start` 让空间排布在所有元素末尾。使用 `flex-end` 让空间排布在所有元素之前，`center` 可以居中导航元素。
 
 {{EmbedGHLiveSample("css-examples/flexbox/use-cases/navigation.html", '100%', 550)}}
 
 ### 让元素自己处理空间分布
 
-导航的另一个不同模式是让元素自己去决定如何处理额外的空间，而不是将空间分布在它们之间。在这种情况下，我们使用 {{cssxref("flex")}} 属性来允许各元素彼此成比例的拉伸和收缩，正如 [Controlling ratios of flex items along the main axis](/zh-CN/docs/Web/CSS/CSS_flexible_box_layout/Controlling_ratios_of_flex_items_along_the_main_axis) 所描述。
+导航的另一个不同模式是让元素自己去决定如何分分布额外的空间，而不是将空间分布在它们之间。在这种情况下，我们使用 {{cssxref("flex")}} 属性来允许各元素彼此成比例的拉伸和收缩，正如[控制弹性元素在主轴上的比例](/zh-CN/docs/Web/CSS/CSS_flexible_box_layout/Controlling_ratios_of_flex_items_along_the_main_axis)所描述的。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md:34]:
```suggestion
导航的另一个不同模式是让元素自己去决定如何分布额外的空间，而不是将空间分布在它们之间。在这种情况下，我们使用 {{cssxref("flex")}} 属性来允许各元素彼此成比例的拉伸和收缩，正如[控制弹性元素在主轴上的比例](/zh-CN/docs/Web/CSS/CSS_flexible_box_layout/Controlling_ratios_of_flex_items_along_the_main_axis)所描述的。
```

File: files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md [hunk 3]
```diff
@@ -1,89 +1,93 @@
...
 
-如果我想让导航中的所有元素都等宽，会使用 `flex: auto`，这是 `flex: 1 1 auto` 的简写形式。所有元素都在它们的 flex-basis 尺寸上进行自动的收缩。这意味着，较长的元素会获得更多的空间。因为 flex-basis 的值被设置为 0，所以所有空间都会被平均分配。
+如果你想要遵循导航元素的 size 属性，但又要在它们之间平均分配额外的空间，那么你可以使用 `flex: auto`，这是 `flex: 1 1 auto` 的简写——所有元素都在它们的 flex-basis 的基础上自动伸缩。这意味这较长的元素将具有更多的空间，因为它从一个较大的尺寸开始，即使它们都分配了相同的可用空间。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md:36]:
```suggestion
如果你想要遵循导航元素的 size 属性，但又要在它们之间平均分配额外的空间，那么你可以使用 `flex: auto`，这是 `flex: 1 1 auto` 的简写——所有元素都在它们的 flex-basis 的基础上自动伸缩。这意味着较长的元素将具有更多的空间，因为它从一个较大的尺寸开始，即使它们都分配了相同的可用空间。
```

File: files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md [hunk 4]
```diff
@@ -1,89 +1,93 @@
...
+弹性盒就能解决这个问题。我们为卡片创建一个弹性容器，并启用 {{cssxref("flex-direction")}}`: column`。然后我们在为内容区域启用 `flex: 1`（`flex: 1 1 0` 的缩略形式），这个元素就可以在 `flex-basis` 为 0 的基础上伸缩。因为这是唯一一个可以伸展的元素，它会占据所有在弹性容器中可以占据的空间，同时将页脚推至底部。如果你移除示例中的 `flex` 属性你就会看见页脚回到了内容的底部。
 
 {{EmbedGHLiveSample("css-examples/flexbox/use-cases/cards.html", '100%', 800)}}
 
 ## 媒体对象
 
 媒体对象是网页设计中的常见模式：这种模式下，一侧具有图片或其他元素，另一侧具有文本。理想情况下，媒体对象应该可以翻转：即把图片从左侧移动到右侧。
 
-这种模式随处可见，用于评论、以及其他需要显示图片和描述的地方。使用 flexbox 可以允许包含图片的媒体对象部分从图片中获取其尺寸调整信息，并对媒体对象的主体进行弹性布局，以占用剩余空间。
+这种模式随处可见，用于评论以及其他需要显示图片和描述的地方。使用弹性盒子可以允许包含图片的媒体对象部分获取图片的尺寸调整信息，并对媒体对象的主体进行弹性布局，以占用剩余空间。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md:74]:
```suggestion
这种模式随处可见，用于评论以及其他需要显示图片和描述的地方。使用弹性盒子，我们可以让媒体对象中包含图片的部分从图片中获取尺寸信息，并对媒体对象的主体进行弹性布局，以占用剩余空间。
```

File: files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md [hunk 5]
```diff
@@ -1,89 +1,93 @@
...
 
-在下面的实例中，你可以看到我们的媒体对象。使用对齐属性来将交叉轴上的元素对齐到`flex-start`，然后为`.content` flex 元素设置为`flex: 1`。与上面的列布局卡片模式一样，启用`flex: 1`表示此部分卡片可以延伸。
+在下面的实例中，你可以看到我们的媒体对象。使用对齐属性来将交叉轴上的元素对齐到 `flex-start`，然后为 `.content` 弹性元素设置为 `flex: 1`。与上面的列布局卡片模式一样，启用 `flex: 1` 表示这部分卡片可以延伸。
```
Comment (jasonren0403) [inline files/zh-cn/web/css/css_flexible_box_layout/typical_use_cases_of_flexbox/index.md:76]:
```suggestion
在下面的实时示例中，你可以看到我们的媒体对象。使用对齐属性来将交叉轴上的元素对齐到 `flex-start`，然后为 `.content` 弹性元素设置为 `flex: 1`。与上面的列布局卡片模式一样，启用 `flex: 1` 表示这部分卡片可以延伸。
```

**Review by jasonren0403 — CHANGES_REQUESTED** [review]
some suggestions

**Review by jasonren0403 — APPROVED** [review]
Thank you @yin1999 👍

**Pull request merged**